### PR TITLE
Add base conversion module

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: six/modules/c++/samples/round_trip_six.cpp [hunk 1]
```diff
@@ -190,6 +191,9 @@ int main(int argc, char** argv)
                            "level", "LEVEL")->setChoices(
                            str::split("debug info warn error"))->setDefault(
                            "info");
+        parser.addArgument("-p --plugin",
```
Comment (asylvest) [inline six/modules/c++/samples/round_trip_six.cpp]: Spell out in the help message what these plugins are for

File: six/modules/c++/six.convert/include/six/convert/ConverterProvider.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+#include <six/sicd/ComplexData.h>
+#include <sys/Conf.h>
+#include <types/RowCol.h>
+
+namespace six
+{
+namespace convert
+{
+class ConverterProvider
```
Comment (asylvest) [inline six/modules/c++/six.convert/include/six/convert/ConverterProvider.h]: Add Doxygen for what this class does

File: six/modules/c++/six.convert/include/six/convert/ConverterProvider.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    virtual void load(const std::string& pathname) = 0;
+
+    //! Parse loaded file and return a ComplexData object
+    virtual std::auto_ptr<six::sicd::ComplexData> convert() = 0;
+
+    /*!
+     * Read given region from image
+     * \param startingLocations Starting row and column
+     * \param readLengths Number of rows and columns to reade
+     * \param buffer Pre-allocated buffer for storing image data
```
Comment (asylvest) [inline six/modules/c++/six.convert/include/six/convert/ConverterProvider.h]: Change readLengths to dims to be consistent with other naming conventions

Comment (asylvest) [inline six/modules/c++/six.convert/include/six/convert/ConverterProvider.h]: Add to this note how many bytes buffer must be

Comment (asylvest) [inline six/modules/c++/six.convert/include/six/convert/ConverterProvider.h]: Make buffer void* here and below so the caller can have a buffer of whatever type they want

File: six/modules/c++/six.convert/include/six/convert/ConvertingReadControl.h [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+#include <six/Enums.h>
+#include <six/ReadControl.h>
+#include <six/convert/ConverterProvider.h>
+#include <six/convert/ConverterProviderRegistry.h>
+
+namespace six
+{
+namespace convert
+{
+class ConvertingReadControl : public ReadControl
```
Comment (asylvest) [inline six/modules/c++/six.convert/include/six/convert/ConvertingReadControl.h]: Add Doxygen for class description

**Comment by JonathanMeans** [comment]
Done.

File: six/modules/c++/six.convert/wscript [hunk 5]
```diff
@@ -0,0 +1,14 @@
+NAME            = 'six.convert'
+MAINTAINER      = 'adam.sylvester@mdaus.com'
```
Comment (asylvest) [inline six/modules/c++/six.convert/wscript]: You are the maintainer :o)

File: six/modules/c++/six.convert/wscript [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+MODULE_DEPS     = 'cli io plugin six.sicd sys xml.lite'
+UNITTEST_DEPS   = 'cli sio.lite'
```
Comment (asylvest) [inline six/modules/c++/six.convert/wscript]: Don't need UNITTEST_DEPS or the `import os` line

File: six/modules/c++/six.convert/wscript [hunk 7]
```diff
@@ -0,0 +1,14 @@
+NAME            = 'six.convert'
+MAINTAINER      = 'adam.sylvester@mdaus.com'
+MODULE_DEPS     = 'cli io plugin six.sicd sys xml.lite'
```
Comment (asylvest) [inline six/modules/c++/six.convert/wscript]: MODULE_DEPS doesn't need `cli` does it?

**Comment by JonathanMeans** [comment]
Everything's fixed.

**Comment by asylvest** [comment]
Looks good

**Pull request merged**